### PR TITLE
Feature/wappu radio

### PR DESCRIPTION
**Pull request by kaupunki-apina**
- New endpoint listing available radio stations

### CODE REVIEW

File: migrations/20170314082605_add_radio_table.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('radios', function(table) {
+    table.increments('id').primary().index();
+    table.string('name').notNullable().index();
+    table.string('stream');
```
Comment (tomi) [inline migrations/20170314082605_add_radio_table.js]: Wondering if this should be NOT NULL.. Does it make sense to have a radio without a stream?

File: README.md [hunk 2]
```diff
@@ -305,6 +305,20 @@ Responses:
...
 
 
+### `GET /api/radio`
+
+> Get list of radio stations.
+
+Query parameters:
+
+* `cityName` String. If specified, returns only stations active in the given city.
+* `cityId` Integer. If specified, returns only stations active in the given city.
```
Comment (tomi) [inline README.md]: Do we need to be able to filter with both id and name? Would id be enough?

File: seeds/radios.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    rows.forEach(city => {
+      cities[city.name] = city.id;
+    });
+  })
+  .then(() => util.insertOrUpdate(knex, 'radios', {
+    id: 1,
+    name: 'Rakkauden Wappuradio',
+    city_id: cities['Tampere'],
+    stream: null,   // TODO
+    website: null,  // TODO
```
Comment (tomi) [inline seeds/radios.js]: For wappuradio the website is https://wappuradio.fi/

File: src/core/radio-core.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        deepChangeKeyCase(row, 'camelCase')
+      )
+    );
+}
+
+function _getWhereClause(filters) {
+  let whereClauses = {};
+
+  if (filters.cityId) whereClauses.city_id = filters.cityId;
+  if (filters.radioId) whereClauses.id = filters.radioId;
```
Comment (tomi) [inline src/core/radio-core.js]: We probably don't have an eslint rule to enforce curly braces in `if` statements but that has been the convention

File: src/http/radio-http.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import * as radioCore from '../core/radio-core';
+import {createJsonRoute} from '../util/express';
+import {assert} from '../validation';
+
+const getStations = createJsonRoute(function(req, res) {
+  const params = assert({
+    radioId: req.query.radioId,
+    cityId: req.query.cityId,
+    cityName: req.query.cityName,
+  }, 'radioParams');
```
Comment (tomi) [inline src/http/radio-http.js]: Isn't this the same as `const params = assert(req.query, 'radioParams');`?

Comment (kaupunki-apina) [inline src/http/radio-http.js]: Effectively yes. But in feed-http similar convention was used.

File: README.md (hunk 2 above)
Comment (kaupunki-apina) [inline README.md]: Other end points capable of filtering results by city can do so by name or id. For consistency's sake, filtering by name should be dropped from other endpoints as well should we choose to drop it from this endpoint.

File: migrations/20170314082605_add_radio_table.js (hunk 1 above)
Comment (kaupunki-apina) [inline migrations/20170314082605_add_radio_table.js]: Its a question of will the radio streams be available upon launch.

**Pull request merged**